src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1356,11 +1356,39 @@ MavlinkReceiver::handle_message_set_attitude_target(mavlink_message_t *msg)
 						att_sp.yaw_sp_move_rate = 0.0f;
 					}
 
-					// TODO: We assume offboard is only used for multicopters which produce thrust along the
-					// body z axis. If we want to support fixed wing as well we need to handle it differently here, e.g.
-					// in that case we should assign att_sp.thrust_body[0]
 					if (!_offboard_control_mode.ignore_thrust) { // dont't overwrite thrust if it's invalid
-						att_sp.thrust_body[2] = -set_attitude_target.thrust;
+						// Fill correct field by checking frametype
+						// TODO: add as needed
+						switch (_mavlink->get_system_type()) {
+						case MAV_TYPE_GENERIC:
+							break;
+
+						case MAV_TYPE_FIXED_WING:
+							break;
+
+						case MAV_TYPE_QUADROTOR:
+							att_sp.thrust_body[2] = -set_attitude_target.thrust;
+							break;
+
+						case MAV_TYPE_HELICOPTER:
+							break;
```

Comment:
You can group all of the different rotary wing options together.

case MAV_TYPE_QUADROTOR:
case MAV_TYPE_HELICOPTER:
case MAV_TYPE_HEXAROTOR:
case MAV_TYPE_OCTOROTOR:
case MAV_TYPE_TRICOPTER:
    att_sp.thrust_body[2] = -set_attitude_target.thrust;

Reply:
Helicopters as well? I think they have great difference, are n' t they?

Reply:
Yes, but in this context we're simply talking about Mavlink SET_ATTITUDE_TARGET (attitude + thrust). Right now we only need to distinguish between thrust down (helicopters, quadcopters, hex, etc) and thrust backwards (planes and cars).